src/execution/operator/helper/physical_set.cpp

```diff
@@ -7,13 +9,37 @@ namespace duckdb {
 void PhysicalSet::GetChunkInternal(ExecutionContext &context, DataChunk &chunk, PhysicalOperatorState *state) const {
 	D_ASSERT(scope == SetScope::GLOBAL || scope == SetScope::SESSION);
 
+	auto normalized_name = ValidateInput(context);
 	if (scope == SetScope::GLOBAL) {
-		context.client.db->config.set_variables[name] = value;
+		context.client.db->config.set_variables[normalized_name] = value;
 	} else {
-		context.client.set_variables[name] = value;
+		context.client.set_variables[normalized_name] = value;
 	}
 
 	state->finished = true;
 }
 
+string PhysicalSet::ValidateInput(ExecutionContext &context) const {
+	if (name == "search_path" || name == "schema") {
+		auto paths = StringUtil::Split(value.str_value, ',');
+
+		// The PG doc says:
+		// >  SET SCHEMA 'value' is an alias for SET search_path TO value.
+		// >  Only one schema can be specified using this syntax.
+		if (name == "schema" && paths.size() > 1) {
+			throw CatalogException("SET schema can set only 1 schema. This has %d", paths.size());
+		}
+
+		for (const auto &path : paths) {
+			if (!context.client.db->GetCatalog().GetSchema(context.client, StringUtil::Lower(path), true)) {
```

Comment:
Explicit `StringUtil::Lower` should not be required here as the catalog uses case insensitive lookups.